engine/userguide/storagedriver/overlayfs-driver.md

```diff
@@ -4,412 +4,470 @@ keywords: container, storage, driver, OverlayFS
-also be faster than `btrfs`. That said, there are a few things to be aware of
-relative to the performance of Docker using the `overlay`/`overlay2` storage
-drivers.
+Both `overlay2` and `overlay` drivers are more performant than `aufs` and
+`devicemapper`. In certain circumstances, `overlay2` may perform better than
+`btrfs` as well. However, be aware of the following details.
 
-- **Page Caching**. OverlayFS supports page cache sharing. This means multiple
-containers accessing the same file can share a single page cache entry (or
-entries). This makes the `overlay`/`overlay2` drivers efficient with memory and
-a good option for PaaS and other high density use cases.
+- **Page Caching**. OverlayFS supports page cache sharing. Multiple containers
+  accessing the same file share a single page cache entry for that file. This
+  makes the `overlay` and `overlay2` drivers efficient with memory and a good
+  option for high-density use cases such as PaaS.
 
-- **copy_up**. As with AUFS, OverlayFS has to perform copy-up operations any
-time a container writes to a file for the first time. This can insert latency
-into the write operation &mdash; especially if the file being copied up is
-large. However, once the file has been copied up, all subsequent writes to that
- file occur without the need for further copy-up operations.
+- **copy_up**. As with AUFS, OverlayFS has to perform copy-up operations
+  whenever a container writes to a file for the first time. This can add latency
+  into the write operation, especially for large files. However, once the file
+  has been copied up, all subsequent writes to that file occur in the upper
+  layer, without the need for further copy-up operations.
 
-    The OverlayFS copy_up operation should be faster than the same operation
-with AUFS. This is because AUFS supports more layers than OverlayFS and it is
-possible to incur far larger latencies if searching through many AUFS layers.
+  The OverlayFS `copy_up` operation is faster than the same operation with AUFS,
+  because AUFS supports more layers than OverlayFS and it is possible to incur
+  far larger latencies if searching through many AUFS layers. `overlay2` supports
+  multiple layers as well, but mitigates any performance hit with caching.
 
 - **Inode limits**. Use of the `overlay` storage driver can cause excessive
-inode consumption. This is especially so as the number of images and containers
- on the Docker host grows. A Docker host with a large number of images and lots
- of started and stopped containers can quickly run out of inodes. The `overlay2`
- does not have such an issue.
+  inode consumption. This is especially true in the presence of a large number
+  of images and containers on the Docker host. The only way to increase the
+  number of inodes available to a filesystem is to reformat it. To avoid running
+  into this issue, it is highly recommended that you use `overlay2` if at all
```

Comment:
For EE, we are specifically only supporting the configs in https://success.docker.com/Policies/Compatibility_Matrix.